Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp

```diff
@@ -310,7 +310,7 @@ namespace AzToolsFramework
 
             static const char* QUERY_PRODUCT_BY_JOBID = "AzToolsFramework::AssetDatabase::QueryProductByJobID";
             static const char* QUERY_PRODUCT_BY_JOBID_STATEMENT =
-                "SELECT Products.*, Jobs.* FROM Jobs INNER JOIN Products "
+                "SELECT Products.*, Jobs.* FROM Jobs LEFT JOIN Products "
```

Comment:
Author's note, this one was a crazy one to find.  This query was using an INNER join, which causes it to drop any rows that have no products in it, but the places the query was being USED was places where it was looking for the source, given the JobID.  

This caused any jobs which emit no products to be omitted.

a LEFT join causes it to still include those lines, just with blanks for products, if none are found.